parser/protobuf.js

```diff
@@ -24,30 +27,25 @@ parseProtobufString = function (protoBufString, options, loaded) {
     options = optionsHandler.verify(options);
     var parser = new ProtoBuf.DotProto.Parser(protoBufString);
     var data = parser.parse();
-    loaded = loaded || {};
-    if (Array.isArray(data.imports)) {
-        var imports = data.imports;
-        for (var i = 0; i < imports.length; i++) {
+    loaded = loaded || [];
+    if (Array.isArray(data['imports'])) {
+        var imports = data['imports'];
+        for (var i=0; i<imports.length; i++) {
             // Skip pulled imports and legacy descriptors
-            if (typeof imports[i] !== 'string' || (util.isDescriptor(imports[i]) && !options.legacy)) {
+            if (typeof imports[i] !== 'string' || (util.isDescriptor(imports[i]) && !options.legacy))
                 continue;
-            }
             // Merge imports, try include paths
-            var fileFound = false;
-            var path = options.path || [];
-            for (var j = 0; j < path.length; ++j) {
-                var importFilename = nodePath.resolve(path[j] + '/', imports[i]);
-                if (!fs.existsSync(importFilename)) {
-                    continue;
+            (function() {
+                var path = options.path || [];
```

Comment:
The path viarable here is overwrites the outer scope "path" variable which imports the node path module.